### ISSUE TITLE
the reference line for RelativeLanePosition or RelativeRoadPosition during TrajecctoryAction

### ISSUE THREAD

**Issue by bleach31** [esmini/esmini#267]
When executing a TeleportAction to override a TrajecctoryAction, the reference line becomes a trajectory.

```xml
<TeleportAction>
<Position>
   <RelativeLanePosition entityRef="Ego" dLane="0" ds="0" offset="0.0"></RelativeLanePosition>
   <!-- <RelativeRoadPosition entityRef="Ego" ds="0.0" dt="0.0" />-->
</Position>
</TeleportAction>
````

In the above example, the vehicle does not move to the centre of the lane (or road).

 It simply cancels the trajectory action.

If this behaviour is expected, TrajectoryPosition can be used.

Is this the correct behaviour of OpenSCENARIO?

And I expect Esmini to make an assignment to the nearest lane/road and use it as a reference line, as if the trajectory operation had been completed normally

Please consider that.
[20220404_RRP_offset.xosc.txt](https://github.com/esmini/esmini/files/8408256/20220404_RRP_offset.xosc.txt)

**Comment by eknabevcc** [comment]
Indeed, you're right! The offset attribute of RelativeLanePosition should be absolute (to the center of the final lane), not relative to the offset of the related position.

So it is a nice difference from the RelativeRoadPosition, where dt should be relative to the t-value of related position.
This way you get two different behaviors that can serve different use cases.

Very good finding. Thanks for reporting and providing the scenario for our reproduction. 👍

We have updated the code accordingly, by commit 9da8a3bc33b438da46d0495ab7567d65ecae24c5.
It's included in release [2.22.1](https://github.com/esmini/esmini/releases/tag/v2.22.1)

It will probably cause some short term confusion for some users, but it's a good fix long term.

Please test and report if the updated behavior is as expected or not.

**Comment by bleach31** [comment]
Thank you so much.

I have confirmed the expected behaviour.
